Code/autopkglib/SparkleUpdateInfoProvider.py

```diff
@@ -181,59 +235,64 @@ def get_feed_data(self, url):
             enclosure = item_elem.find("enclosure")
             if enclosure is not None:
                 item = {}
-                # URL-quote the path component to handle spaces, etc.
-                # (Panic apps do this)
-                url_bits = urlparse.urlsplit(enclosure.get("url"))
-                if self.env.get("urlencode_path_component", True):
-                    encoded_path = urllib.quote(url_bits.path)
-                else:
-                    encoded_path = url_bits.path
-                built_url = url_bits.scheme + "://" + url_bits.netloc + encoded_path
-                if url_bits.query:
-                    built_url += "?" + url_bits.query
-                item["url"] = built_url
-
-                item["version"] = enclosure.get("{%s}version" % xmlns)
-                if item["version"] is None:
-                    # Sparkle tries to guess a version from the download URL for
-                    # rare cases where there is no sparkle:version enclosure
-                    # attribute, for the format: AppnameOrAnything_1.2.3.zip
-                    # https://github.com/sparkle-project/Sparkle/blob/
-                    # 89081ca030c0de218400f7c0f97530df524d687d/Sparkle/
-                    # SUAppcastItem.m#L69-L76
-                    #
-                    # We can even support OmniGroup's alternate appcast format
-                    # by cheating and using the '-' as a delimiter to derive
-                    # version info
-                    filename = os.path.basename(os.path.splitext(item["url"])[0])
-                    for delimiter in ["_", "-"]:
-                        if delimiter in filename:
-                            item["version"] = filename.split(delimiter)[-1]
-                            break
-                # if we still found nothing, fail
-                if item["version"] is None:
-                    raise ProcessorError(
-                        "Can't extract version info from item in feed!"
-                    )
+                item["url"] = self.build_url(enclosure)
+                item["version"] = self.determine_version(enclosure, item["url"])
 
-                human_version = enclosure.get("{%s}shortVersionString" % xmlns)
+                human_version = enclosure.get("{%s}shortVersionString" % self.xmlns)
```

Comment:
Not sure I'd necessarily state that this is the "human version" because it depends heavily on the vendor, but not a blocker.

Reply:
What alternate name do you suggest?